### DISCUSSION TITLE
Local Repo file / Folder not found

### DISCUSSION THREAD

**Discussion by hsh-it** [semaphoreui/semaphore#1724] [Q&A] [answered]
Hey, i installed Semaphore as root with snap
```
Linux 5.15.0-92-generic #102-Ubuntu SMP Wed Jan 10 09:33:48 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

22.04.3 LTS
$ semaphore version
v2.9.4
$ ansible --version
ansible [core 2.15.4]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3/dist-packages/ansible
  ansible collection location = /root/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] (/usr/bin/python3)
  jinja version = 3.0.3
  libyaml = True
```
Semaphore runs as root
```
$ ps aux | grep semaphore
root        1090  0.0  0.0   7636  5176 ?        Ss   Jan26   0:01 /bin/bash /snap/semaphore/260/semaphored
root        1628  0.0  0.3 747812 61184 ?        Sl   Jan26   0:21 semaphore service --config /root/snap/semaphore/common/config.json
sysadm    159577  0.0  0.0   7004  2224 pts/0    S+   08:29   0:00 grep --color=auto semaphore
```

I want to change from CLI version to [semaphore](https://github.com/ansible-semaphore/semaphore).

Path to my Playbooks:
/opt/HSH/ansible_project/ansible-playbooks (Path and files exist!)

Local Repo:
![image](https://github.com/ansible-semaphore/semaphore/assets/150814264/a82e86cc-1ca7-4e49-b17a-6fd705ed18be)

Environment (none)
![image](https://github.com/ansible-semaphore/semaphore/assets/150814264/2870c707-4109-492d-a040-9cc60621c388)

Inventory files created:
![image](https://github.com/ansible-semaphore/semaphore/assets/150814264/d8a8bc1b-309c-4ec0-9fff-377dbd6bc299)

Playbook-File:
```
/opt/HSH/ansible_project/ansible-playbooks $ cat AP-bugfix.yml
---
- name: Run ebtables commands on all hosts
  hosts: unifi_ap
[...]
```

Inventory File:
![image](https://github.com/ansible-semaphore/semaphore/assets/150814264/294567aa-b9d0-44f7-8c43-6361bb8a3452)

I create a new Task:
![image](https://github.com/ansible-semaphore/semaphore/assets/150814264/85e1f33b-9b2e-44eb-9f47-1f5183cd42e3)

Run Task:
![image](https://github.com/ansible-semaphore/semaphore/assets/150814264/faf22c57-e7fc-4563-bcce-f274e4465833)

Here is finaly my issue:
![image](https://github.com/ansible-semaphore/semaphore/assets/150814264/65f9f01b-6a19-44c0-a3a0-520a3b9b7fb6)

```

9:04:19 AM
Task 2147483636 added to queue
9:04:20 AM
Started: 2147483636
9:04:20 AM
Run TaskRunner with template: FIX-APs_new
9:04:20 AM
Preparing: 2147483636
9:04:20 AM
Failed in finding static repository at /opt/HSH/ansible_project/ansible-playbooks: stat /opt/HSH/ansible_project/ansible-playbooks: no such file or directory
9:04:20 AM
Running playbook failed: stat /opt/HSH/ansible_project/ansible-playbooks: no such file or directory
```

Any Idea how to solve it ?

**Comment by canlot** [comment]
Had the same problem with the installation over snap, i found somewhere in the internet that snap packages can only access certain snap paths.
You can use git or like me install it manually.

**Comment by fiftin** [comment, marked as answer]
Hello,

Snap without `--classic` option allows access only to Home directory. Try to move playbooks to Home directory.

Semaphore's snap works without `--classic` for security reason because snap service works under root user.

**Reply by hsh-it** [reply]
Whee i find the home directory ?
`/root/snap/semaphore/common/ansible-playbook`
